src/frontend/src/icp-eth/components/tokens/IcManageTokens.svelte

```diff
@@ -69,40 +70,47 @@
 			);
 
 	let tokens: IcrcCustomToken[] = [];
-	$: tokens = filteredTokens.map(({ ledgerCanisterId, enabled, ...rest }) => ({
-		ledgerCanisterId,
-		enabled: modifiedTokens[`${ledgerCanisterId}`]?.enabled ?? enabled,
+	$: tokens = filteredTokens.map(({ id, network, enabled, ...rest }) => ({
+		id,
+		network,
+		enabled: modifiedTokens[network.id]?.[id]?.enabled ?? enabled,
 		...rest
 	}));
 
 	let noTokensMatch = false;
 	$: noTokensMatch = tokens.length === 0;
 
-	let modifiedTokens: Record<CanisterIdText, IcrcCustomToken> = {};
-	const onToggle = ({
-		detail: { ledgerCanisterId, enabled, ...rest }
-	}: CustomEvent<IcrcCustomToken>) => {
-		const { [`${ledgerCanisterId}`]: current, ...tokens } = modifiedTokens;
+	let modifiedTokens: Record<NetworkId, Record<TokenId, IcrcCustomToken>> = {};
+	const onToggle = ({ detail: { id, network, ...rest } }: CustomEvent<IcrcCustomToken>) => {
+		const { id: networkId } = network;
 
-		if (nonNullish(current)) {
-			modifiedTokens = { ...tokens };
+		modifiedTokens[networkId] = modifiedTokens[networkId] || {};
+
+		if (nonNullish(modifiedTokens[networkId][id])) {
+			delete modifiedTokens[networkId][id];
```

Comment:
Is there a way to avoid the use of `delete` - i.e. implement a more functional approach? 

A bit an opiniated feedback but, somehow I remember having read some articles explaining that `delete` was a bad practice.

If it would become too verbose let's keep those but, generally I try to avoid those.

Reply:
yes, removing it!

in another PR, I will add the lint rule eheh